shared/middleware/files_api.rb

```diff
@@ -232,6 +232,21 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
     put_file('assets', encrypted_channel_id, file[:filename], file[:tempfile].read)
   end
 
+  # POST /v3/assets/<channel-id>/copy?src_channel=<src-channel-id>&src_files=<src-filenames-json>
+  #
+  # Copy assets from another channel. Note that when specifying the src files, you must
+  # json encode it
+  #
+  post %r{/v3/assets/([^/]+)/copy$} do |encrypted_channel_id|
+    dont_cache
+
+    AssetBucket.new.copy_files(
+      request.GET['src_channel'],
+      encrypted_channel_id,
+      JSON.parse(request.GET(['src_filenames']))
+    )
+  end
```

Comment:
I would do it like this: `POST /v3/copy-assets/<channel-id>?src_channel=<src-channel-id>&src_files=<src-filenames-json>`. This falls into the "do bulk operations by posting to an entirely separate resource" strategy. The other way that would make sense to me would be to do a PATCH on the collection, e.g. `PATCH /v3/assets/<channel-id>?command=copy&src_channel=<src-channel-id>&src_files=<src-filenames-json>`, which seems the most RESTful of all to me. I can't find any articles confirming that the second way is a good idea, so perhaps you should go with the first.

Regarding the way you were doing it, it can actually be ok to have nested resources, but since `/v3/assets/<channel-id>` is a collection, it looks like you are POSTing to an item in that collection named `copy`, which doesn't seem right.